server/controllers/users/post.js

```diff
@@ -12,12 +13,13 @@ exports.post = (req, res, next) => {
     .then((results) => {
       if (results.rowCount) {
         return res.status(422).send({ error: { code: 422, msg: 'UserName is already taken' } });
-      } return true;
-    })
-    .then(() => insertUser(req.body))
-    .then((response) => {
-      if (response.rowCount) return res.send({ error: null, data: response.rows[0] });
-      return next({ code: 422 });
+      }
+      insertUser(req.body)
```

Comment:
We agreed to make it so because she wants to send different two error msgs
1- validation error 422
2- other server error 500
which as I know of can't be achieved without nesting.